administrator/components/com_fields/models/fields/type.php

```diff
@@ -91,4 +107,31 @@ protected function getOptions ()
 
 		return $options;
 	}
+
+	/**
+	 * Parses the file with the given path. If it is a class starting with the
+	 * name JFormField and implementing JFormDomfieldinterface, then the class name is returned.
+	 *
+	 * @param string $path
+	 * @return string|boolean
+	 */
+	private function getClassNameFromFile($path)
+	{
+		$tokens = token_get_all(JFile::read($path));
```

Comment:
The problem was that many form fields do extend from JFormFieldList (which do not make sense to be available on com_fields) and would pass the check here https://github.com/joomla-projects/custom-fields/pull/104/files/6ad70761515e61fc42d69a6225f18f5c0b34d394#diff-7acc4b763de148d363285b347d14a2f9R615. This function parses the file and checks if the file itself implements JFormDomfieldinterface.
Honestly I'm not very happy with that solution, but it was the only approach which worked at the end. If you have a better solution, then I'm happy to change.